lib/matplotlib/tests/test_figure.py

```diff
@@ -1603,3 +1603,11 @@ def test_savefig_metadata(fmt):
 def test_savefig_metadata_error(fmt):
     with pytest.raises(ValueError, match="metadata not supported"):
         Figure().savefig(io.BytesIO(), format=fmt, metadata={})
+
+
+def test_get_constrained_layout_pads():
+    expected = (0.01, 0.02, 0.03, 0.04)
+    params = dict(zip(['w_pad', 'h_pad', 'wspace', 'hspace'], expected))
```

Comment:
The inverse seems clearer to me:
```suggestion
    params = {'w_pad': 0.01, 'h_pad': 0.02, 'wspace': 0.03, 'hspace': 0.04}
    expected = [*params.values()]
```

Reply:
@QuLogic I'm neutral on this: my understanding of unit testing is that expected values should be immediate values, not derived ones, because we want to test the function under test and not the code for obtaining the expected value. So on second thought the canonical solution in my opinion would have been

```
def test_get_constrained_layout_pads():
    params = {'w_pad': 0.01, 'h_pad': 0.02, 'wspace': 0.03, 'hspace': 0.04}
    fig = plt.figure(layout=mpl.layout_engine.ConstrainedLayoutEngine(**params))
    with pytest.warns(PendingDeprecationWarning, match="will be deprecated"):
        assert fig.get_constrained_layout_pads() == (0.01, 0.02, 0.03, 0.04)
```

(but you were too quick to merge, no problem)